src/map_name_popup.c

```diff
@@ -0,0 +1,452 @@
+        }
+        else
+        {
+            if (gTasks[mapNamePopupTaskId].data[0] != 2)
+                gTasks[mapNamePopupTaskId].data[0] = 2;
+            gTasks[mapNamePopupTaskId].data[3] = 1;
+        }
+    }
+}
+
+static void Task_MapNamePopUpWindow(u8 taskId)
+{
+    struct Task *task = &gTasks[taskId];
+
+    switch (task->data[0])
+    {
+    case 6:
+        task->data[4]++;
+        if (task->data[4] > 30)
+        {
+            task->data[0] = 0;
+            task->data[4] = 0;
+            ShowMapNamePopUpWindow();
+        }
+        break;
+    case 0:
+        task->data[2] -= 2;
+        if (task->data[2] <= 0 )
+        {
+            task->data[2] = 0;
+            task->data[0] = 1;
+            gTasks[mapNamePopupTaskId].data[1] = 0;
+        }
+        break;
+    case 1:
+        task->data[1]++;
+        if (task->data[1] > 120 )
+        {
+            task->data[1] = 0;
+            task->data[0] = 2;
+        }
+        break;
+    case 2:
+        task->data[2] += 2;
+        if (task->data[2] > 39)
+        {
+            task->data[2] = 40;
+            if (task->data[3])
+            {
+                task->data[0] = 6;
+                task->data[4] = 0;
+                task->data[3] = 0;
+            }
+            else
+            {
+                task->data[0] = 4;
+                return;
+            }
+        }
+        break;
+    case 4:
+        sub_819746C(GetMapNamePopUpWindowId(), TRUE);
+        task->data[0] = 5;
+        break;
+    case 5:
+        HideMapNamePopUpWindow();
+        return;
+    }
+    SetGpuReg(REG_OFFSET_BG0VOFS, task->data[2]);
+}
+
+void HideMapNamePopUpWindow(void)
+{
+    if (FuncIsActiveTask(Task_MapNamePopUpWindow))
+    {
+        sub_819746C(GetMapNamePopUpWindowId(), TRUE);
+        RemoveMapNamePopUpWindow();
+        SetGpuReg_ForcedBlank(REG_OFFSET_BG0VOFS, 0);
+        DestroyTask(mapNamePopupTaskId);
+    }
+}
+
+static void ShowMapNamePopUpWindow(void)
+{
+    u8 mapDisplayHeader[24];
+    u8 *withoutPrefixPtr;
+    u8 x;
+    const u8* mapDisplayHeaderSource;
+
+    if(InBattlePyramid())
+    {
+        if(gMapHeader.mapDataId == 0x17A)
+        {
+            withoutPrefixPtr = &(mapDisplayHeader[3]);
+            mapDisplayHeaderSource = gBattlePyramid_MapHeaderStrings[7];
+        }
+        else
+        {
+            withoutPrefixPtr = &(mapDisplayHeader[3]);
+            mapDisplayHeaderSource = gBattlePyramid_MapHeaderStrings[gSaveBlock2Ptr->battlePyramidWildHeaderId];
+        }
+        StringCopy(withoutPrefixPtr, mapDisplayHeaderSource);
+    }
+    else
+    {
+        withoutPrefixPtr = &(mapDisplayHeader[3]);
+        GetMapName(withoutPrefixPtr, gMapHeader.regionMapSectionId, 0);
+    }
+    AddMapNamePopUpWindow();
+    LoadMapNamePopUpWindowBg();
+    x = GetStringCenterAlignXOffset(7, withoutPrefixPtr, 80);
+    mapDisplayHeader[0] = 0xFC;
+    mapDisplayHeader[1] = 0x2;
+    mapDisplayHeader[2] = 0x0;
```

Comment:
Use text control constants from `text.h` for these three bytes: `EXT_CTRL_CODE_BEGIN`, `EXT_CTRL_CODE_HIGHLIGHT`, and `TEXT_COLOR_TRANSPARENT`.